nova_vm/src/engine/bytecode/instructions.rs

```diff
@@ -244,10 +244,20 @@ pub enum Instruction {
     /// Take the current result and begin iterating it according to
     /// EnumerateObjectProperties.
     EnumerateObjectProperties,
+    /// Take the current result and call `GetIterator(result, SYNC)`
+    GetIteratorSync,
+    /// Take the current result and call `GetIterator(result, ASYNC)`
+    GetIteratorAsync,
     /// Call [[NextMethod]] of current iterator
     IteratorNext,
+    /// Check "done" boolean of current iterator result object
     IteratorComplete,
+    // Extract "value" of current iterator result object
     IteratorValue,
+    /// Perform CloseIterator on the current iterator
+    IteratorClose,
+    /// Perform AsyncCloseIterator on the current iterator
+    AsyncIteratorClose,
```

Comment:
Yeah, that sounds reasonable. Currently the act of starting an iterator also only works for synchronous iterators, and an asynchronous iterator will need an intermediate await. This will definitely all need some instruction work and massaging in the future.

Let's leave it here for now as a marker of sorts, though.

Side note: That'd be a **suggestion** or **issue (nonblocking)** maybe? :)